lib/zephyr/src/zephyr/readers.py

```diff
@@ -20,12 +20,50 @@
 from __future__ import annotations
 
 import fnmatch
+import logging
 import zipfile
 from collections.abc import Iterator
 from contextlib import contextmanager
+from dataclasses import dataclass, field
+from typing import Literal
 
 import fsspec
 import msgspec
+import numpy as np
+import pyarrow as pa
+
+from zephyr.expr import Expr
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InputFileSpec:
+    """Specification for reading a file or portion of a file.
+
+    Attributes:
+        path: Path to the file
+        format: File format ("parquet", "jsonl", or "auto" to detect)
+        columns: List of columns to read
+        row_start: Optional start row for chunked reading
+        row_end: Optional end row for chunked reading
+        filter_expr: Optional filter expression to apply
+    """
+
+    path: str
+    format: Literal["parquet", "jsonl", "auto"] = "auto"
+    columns: list[str] | None = None
+    row_start: int | None = None
+    row_end: int | None = None
+    filter_expr: Expr | None = field(default=None)
```

Comment:
Q: why is `field(default=None)` needed here? could this be just `= None`?

Reply:
yes, i think i was trying something fancy originally and this got left in a weird state